### DISCUSSION TITLE
Is there really no HID/USB Joystick built into TinyUSB?

### DISCUSSION THREAD

**Discussion by CCurl** [hathach/tinyusb#864] [Q&A] [answered]
I have to say that I would be somewhat surprised by this if it's true. I see everything else ... Mouse, Keyboard, Gamepad. No Joystick? A Joystick with 6 axes and 32 buttons would be great.

Thanks

**Comment by hathach** [comment]
It is just a matter of adding HID report descriptor and testing. Would you tell me how would you test an joystick, I will try to add when having time.

**Reply by CCurl** [reply]
Wow ... thanks! I would be using it as an extra input device for DCS (the aircraft simulator) to be able to make 'real life' versions of the components in the cockpit. It would also be nice (I don't even know if it's possible though) to be able to name the device, as there are so many switches and analog devices in a cockpit, that I will end up needing multiple of these things.

**Reply by hathach** [reply]
@CCurl I mean normally which application you are using to test the function of the joystick, that just makes sure it would work when you try it out.

**Reply by CCurl** [reply]
I have a Forth system that I wrote that runs on the board. I can use that interactively on the board. On the host (Windows 10), I use 'joy.cpl' on Windows to see if it can detect the button presses the buttons and the changes to the axes.

**Reply by CCurl** [reply]
That Forth system is here: 

https://github.com/CCurl/BoardForth. 

I can easily add operations to call any function in a C library.

**Reply by hathach** [reply]
Thanks, I will this out when I add the joystick example in the future.

**Comment by Algolian** [comment]
Start with a 2 axes and get that work. (that basic hid standard) And they you need have multiple version of that to get the
number of axes you need. It should not be that complicated if you don't going to do a force feedback device.

**Comment by mmosca** [comment, marked as answer]
@CCurl The hid_composite example includes a gamepad.
That gives you 6 axis and 16 buttons.

I use it to implement a joystick in https://github.com/mmosca/pico-sbus

I also submitted a pull request to bump the buttons from 16 to 32.
